src/arrayMethodJoin.js

```diff
@@ -4,8 +4,31 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+    let sep = separator;
```

Comment:
I would not create a separate variable `sep` since you already handle `undefined`: `separator = ','` . Also, name `sep` is not descriptive